server/src/main/java/org/apache/iotdb/db/engine/storagegroup/IDTableFlushTimeManager.java

```diff
@@ -0,0 +1,207 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.engine.storagegroup;
+
+import org.apache.iotdb.db.metadata.idtable.IDTable;
+import org.apache.iotdb.db.metadata.idtable.entry.DeviceEntry;
+
+import java.util.Map;
+
+/**
+ * This class manages last time and flush time for sequence and unsequence determination This class
+ * This class is NOT thread safe, caller should ensure synchronization This class not support
+ * upgrade
+ */
+public class IDTableFlushTimeManager implements ILastFlushTimeManager {
+  IDTable idTable;
+
+  public IDTableFlushTimeManager(IDTable idTable) {
+    this.idTable = idTable;
+  }
+
+  // region set
+  @Override
+  public void setLastTimeAll(long timePartitionId, Map<String, Long> lastTimeMap) {
+    for (Map.Entry<String, Long> entry : lastTimeMap.entrySet()) {
+      idTable.getDeviceEntry(entry.getKey()).putLastTimeMap(timePartitionId, entry.getValue());
+    }
+  }
+
+  @Override
+  public void setLastTime(long timePartitionId, String path, long time) {
+    idTable.getDeviceEntry(path).putLastTimeMap(timePartitionId, time);
+  }
+
+  @Override
+  public void setFlushedTimeAll(long timePartitionId, Map<String, Long> flushedTimeMap) {
+    for (Map.Entry<String, Long> entry : flushedTimeMap.entrySet()) {
+      idTable.getDeviceEntry(entry.getKey()).putFlushTimeMap(timePartitionId, entry.getValue());
+    }
+  }
+
+  @Override
+  public void setFlushedTime(long timePartitionId, String path, long time) {
+    idTable.getDeviceEntry(path).putFlushTimeMap(timePartitionId, time);
+  }
+
+  @Override
+  public void setGlobalFlushedTimeAll(Map<String, Long> globalFlushedTimeMap) {
+    for (Map.Entry<String, Long> entry : globalFlushedTimeMap.entrySet()) {
+      idTable.getDeviceEntry(entry.getKey()).setGlobalFlushTime(entry.getValue());
+    }
+  }
+
+  @Override
+  public void setGlobalFlushedTime(String path, long time) {
+    idTable.getDeviceEntry(path).setGlobalFlushTime(time);
+  }
+
+  // endregion
+
+  // region update
+
+  @Override
+  public void updateLastTime(long timePartitionId, String path, long time) {
+    idTable.getDeviceEntry(path).updateLastTimeMap(timePartitionId, time);
+  }
+
+  @Override
+  public void updateFlushedTime(long timePartitionId, String path, long time) {
+    idTable.getDeviceEntry(path).updateFlushTimeMap(timePartitionId, time);
+  }
+
+  @Override
+  public void updateGlobalFlushedTime(String path, long time) {
+    idTable.getDeviceEntry(path).updateGlobalFlushTime(time);
+  }
+
+  @Override
+  public void updateNewlyFlushedPartitionLatestFlushedTimeForEachDevice(
+      long partitionId, String deviceId, long time) {
+    throw new UnsupportedOperationException("IDTableFlushTimeManager doesn't support upgrade");
+  }
+
+  // endregion
+
+  // region ensure
```

Comment:
remove this

Reply:
This is to support structure view in idea